core/src/test/java/uk/gov/gchq/magmacore/service/MagmaCoreServiceTest.java

```diff
@@ -234,4 +234,74 @@ public void testFindByPredicateOnly() {
         });
     }
 
+    /**
+     * Test that case-insensitive searches find the required entities.
+     */
+    @Test
+    public void testFindByPartialSignAndClassCaseInsensitive() throws MagmaCoreException {
+
+        // Create and populate an in-memory database.
+        final MagmaCoreDatabase db = new MagmaCoreJenaDatabase();
+        SignPatternTestData.createSignPattern(db);
+
+        // Use it to create the services
+        final MagmaCoreService service = new MagmaCoreService(db);
+
+        // Create the PointInTime we're looking for
+        final PointInTime now = SpatioTemporalExtentServices.createPointInTime("now");
+        now.addValue(HQDM.ENTITY_NAME, Instant.now().toString());
+
+        // Find the required Things by sign in a transaction.
+        db.begin();
+        final List<? extends Thing> found1 = service.findByPartialSignAndClassCaseInsensitive(
+                "Person1", SignPatternTestData.classOfPersonIri, now);
```

Comment:
Check this covers both cases?
```suggestion
                "person1", SignPatternTestData.classOfPersonIri, now);
```